api/envoy/api/v2/core/health_check.proto

```diff
@@ -95,7 +101,7 @@ message HealthCheck {
     repeated string request_headers_to_remove = 8;
 
     // If set, health checks will be made using http/2.
-    bool use_http2 = 7;
+    CodecClientType codec_client_type = 7;
```

Comment:
This is a breaking change, so we can do this. Please add a new field and tag the existing one as deprecated, thanks.

Reply:
If we're doing a breaking change, should we have a shared enum which we can use in other places?  Even if we don't move the HCM over today.

Reply:
done